// https://github.com/grpc/grpc/issues/9575#issuecomment-293934506
// use the following to ensure relative import syntax:
// python -m grpc_tools.protoc -I proto --python_out=. --grpc_python_out=. proto/service/*.proto 
//

syntax = "proto3";

service Inpainter {
    rpc Ping (Empty) returns (Empty) {}
    rpc StartInpaintingSession (Empty) returns (StartInpaintingSessionResponse) {}
    rpc Inpaint (InpaintRequest) returns (InpaintResponse) {}
    rpc ReportBenchmarks (Empty) returns (ReportBenchmarksResponse) {}
}

message StartInpaintingSessionResponse {
    string SessionId = 1;
}

message InpaintRequest {
    string SessionId = 1;
    // serialized PyTorch Tensor with shape (C, H, W)
    bytes Image = 2;
    // serialized PyTorch Tensor with shape (C, H, W)
    bytes Mask = 3;
}

message InpaintResponse {
    bytes InpaintedFrame = 1;
    int32 InpaintedFrameId = 2;
}

message ReportBenchmarksResponse {
    string BenchmarkReport = 1;
}

message Empty {
}